README.markdown

```diff
@@ -7793,10 +7793,73 @@ tcpsock:setoption
 
 **context:** *rewrite_by_lua&#42;, access_by_lua&#42;, content_by_lua&#42;, ngx.timer.&#42;, ssl_certificate_by_lua&#42;, ssl_session_fetch_by_lua&#42;*
 
-This function is added for [LuaSocket](http://w3.impa.br/~diego/software/luasocket/tcp.html) API compatibility and does nothing for now. Its functionality will be implemented in future.
+This function is added for [LuaSocket](http://w3.impa.br/~diego/software/luasocket/tcp.html) API compatibility and does nothing for now. Its functionality is implemented `v0.10.18`.
 
 This feature was first introduced in the `v0.5.0rc1` release.
 
+The `option` is a string with the option name, and the value depends on the option being set:
+
+* `keepalive`
+
+	Setting this option to true enables sending of keep-alive messages on
+	connection-oriented sockets. Make sure the `connect` function
+	had been called before, for example,
+
+    ```lua
+
+    tcpsock:setoption("keepalive", true)
+    ```
+* `reuseaddr`
+
+	Enabling this option indicates that the rules used in validating addresses
+	supplied in a call to bind should allow reuse of local addresses. Make sure
+	the `connect` function had been called before, for example,
+
+    ```lua
+
+    tcpsock:setoption("reuseaddr", 0)
+    ```
+* `tcp-nodelay`
+
+	Setting this option to true disables the Nagle's algorithm for the connection.
+	Make sure the `connect` function had been called before, for example,
+
+    ```lua
+
+    tcpsock:setoption("tcp-nodelay", true)
+    ```
+* `sndbuf`
+
+	Sets the maximum socket send buffer in bytes. The kernel doubles this value
+	(to allow space for bookkeeping overhead) when it is set using setsockopt().
+	Make sure the `connect` function had been called before, for example,
+
+    ```lua
+
+    tcpsock:setoption("sndbuf", 1024 * 10)
+    ```
+* `rcvbuf`
+
+	Sets the maximum socket receive buffer in bytes. The kernel doubles this value
+	(to allow space for bookkeeping overhead) when it is set using setsockopt. Make
+	sure the `connect` function had been called before, for example,
+
+    ```lua
+
+    tcpsock:setoption("rcvbuf", 1024 * 10)
+    ```
+These options described above are supported in `v0.10.18`, and more options will be implemented in future.
+
+NOTE: Once the option is set, it will become effective until the function `close` is called. So while using the socket from the connetion pool, make sure the option state is just what you need, for example,
```

Comment:
“the function close” => "the connection is closed".

"make sure the oiption state is just what you need" is confusing. I don't understand what it means. This should be rewritten as

"If you know the connection is from the connection pool and all the in-pool connections already have called the `setoption()` method with the desired socket option state, then you can just skip calling `setoption()` again to avoid the overhead of repeated calls."